src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -717,8 +717,16 @@ private void CheckRefReadOnlySymbols(MethodSymbol symbol)
 
         private void CheckNullableSymbols(MethodSymbol symbol)
         {
-            if (symbol.ReturnType.ContainsNullableReferenceTypes() ||
-                symbol.Parameters.Any(p => p.Type.ContainsNullableReferenceTypes()))
+            var location = symbol.Locations.FirstOrDefault();
+            var returnType = symbol.ReturnType;
+            var parameters = symbol.Parameters;
+            if (returnType.ContainsAnnotatedUnconstrainedTypeParameter() ||
+                parameters.Any(p => p.Type.ContainsAnnotatedUnconstrainedTypeParameter()))
+            {
+                TypeSymbolWithAnnotations.ReportAnnotatedUnconstrainedTypeParameter(location, _diagnostics);
```

Comment:
>ReportAnnotatedUnconstrainedTypeParameter [](start = 42, length = 41)

I'm confused. If I read this code correctly, it seems like we're not reporting on the specific location that caused the problem.
Yet, I think this is covered by `NullableT_LocalFunction` which seems to report a specific diagnostic (either on return type or parameter type). 
I don't understand how that works.

Also, it feels strange to report this kind of diagnostic (for lambdas and local functions) during lowering. Could the check be moved into the relevant symbols?

Reply:
Looking at `LocalFunctionSymbol`, I see that the check is performed there. That's why the `NullableT_LocalFunction` test works.
Then I wonder which scenarios/tests hit this code path?

---
In reply to: [205838678](https://github.com/dotnet/roslyn/pull/28804#discussion_r205838678) [](ancestors = 205838678)

Reply:
Good catch. Removed.

---
In reply to: [205839676](https://github.com/dotnet/roslyn/pull/28804#discussion_r205839676) [](ancestors = 205839676,205838678)